pkg/apis/istio/v1alpha2/values/values_types.proto

```diff
@@ -0,0 +1,911 @@
+
+  ResourcesConfig resources = 9;
+
+  string configNamespace = 10;
+
+  CPUTargetUtilizationConfig cpu = 11;
+
+  // 	NodeSelector                     map[string]interface{}      `json:"nodeSelector,omitempty"`
+  string keepaliveMaxServerConnectionAge = 12;
+
+  map<string, string> deploymentLabels = 13;
+
+  // 	MeshNetworks                     map[string]interface{}      `json:"meshNetworks,omitempty"`
+  // 	PodAntiAffinityLabelSelector     []map[string]interface{}    `json:"podAntiAffinityLabelSelector"`
+  // 	PodAntiAffinityTermLabelSelector []map[string]interface{}    `json:"podAntiAffinityTermLabelSelector"`
+  bool configMap = 14;
+
+  PilotIngressConfig ingress = 15;
+
+  bool useMCP = 16;
+
+  map<string, string> env = 17;
+
+  PilotPolicyConfig policy = 18;
+
+  PilotTelemetryConfig telemetry = 19;
+}
+
+// PilotIngressConfig is described in istio.io documentation.
+message PilotIngressConfig {
+  string ingressService = 1;
+
+  string ingressControllerMode = 2;
+
+  string ingressClass = 3;
+}
+
+// PilotPolicyConfig is described in istio.io documentation.
+message PilotPolicyConfig {
+  bool enabled = 1;
+}
+
+// PilotTelemetryConfig is described in istio.io documentation.
+message PilotTelemetryConfig {
+  bool enabled = 1;
+}
+
+// PortsConfig is described in istio.io documentation.
+message PortsConfig {
+  string name = 1;
+
+  int32 port = 2;
+
+  int32 nodePort = 3;
+
+  int32 targetPort = 4;
+}
+
+// PrometheusConfig is described in istio.io documentation.
+message PrometheusConfig {
+  bool createPrometheusResource = 1;
+
+  bool enabled = 2;
+
+  uint32 replicaCount = 3;
+
+  string hub = 4;
+
+  string tag = 5;
+
+  string retention = 6;
+
+  // 	NodeSelector             map[string]interface{}    `json:"nodeSelector,omitempty"`
+  string scrapeInterval = 7;
+
+  string contextPath = 8;
+
+  AddonIngressConfig ingress = 9;
+
+  PrometheusServiceConfig service = 10;
+
+  PrometheusSecurityConfig security = 11;
+}
+
+// PrometheusMixerAdapterConfig is described in istio.io documentation.
+message PrometheusMixerAdapterConfig {
+  bool enabled = 1;
+
+  string metricsExpiryDuration = 2;
+}
+
+// PrometheusSecurityConfig is described in istio.io documentation.
+message PrometheusSecurityConfig {
+  bool enabled = 1;
+}
+
+// PrometheusServiceConfig is described in istio.io documentation.
+message PrometheusServiceConfig {
+  // 	Annotations map[string]interface{}           `json:"annotations,omitempty"`
+  PrometheusServiceNodePortConfig nodePort = 1;
+}
+
+// PrometheusServiceNodePortConfig is described in istio.io documentation.
+message PrometheusServiceNodePortConfig {
+  bool enabled = 1;
+
+  uint32 port = 2;
+}
+
+// ProxyConfig specifies how proxies are configured within Istio.
+message ProxyConfig {
+  // Specifies the path to write the sidecar access log file.
+  string accessLogFile = 1;
+
+  // Configures how and what fields are displayed in sidecar access log.
+  string accessLogFormat = 2;
```

Comment:
enum?

Reply:
the log encoding is enum but this might be not?